recipe/thirdparty/main.go

```diff
@@ -72,27 +73,34 @@ func GetUserByThirdPartyInfo(thirdPartyID, thirdPartyUserID string) (*tpmodels.U
 	return GetUserByThirdPartyInfoWithContext(thirdPartyID, thirdPartyUserID, &map[string]interface{}{})
 }
 
-// TODO enable later
-// func Apple(config tpmodels.AppleConfig) tpmodels.TypeProvider {
-// 	return providers.Apple(config)
-// }
+func ActiveDirectory(input tpmodels.ProviderInput) tpmodels.TypeProvider {
+	return providers.ActiveDirectory(input)
+}
+
+func Apple(input tpmodels.ProviderInput) tpmodels.TypeProvider {
+	return providers.Apple(input)
+}
+
+func Discord(input tpmodels.ProviderInput) tpmodels.TypeProvider {
+	return providers.Discord(input)
+}
 
-// func Facebook(config tpmodels.FacebookConfig) tpmodels.TypeProvider {
-// 	return providers.Facebook(config)
-// }
+func Facebook(input tpmodels.ProviderInput) tpmodels.TypeProvider {
+	return providers.Facebook(input)
+}
 
-// func Github(config tpmodels.GithubConfig) tpmodels.TypeProvider {
-// 	return providers.Github(config)
-// }
+func Github(input tpmodels.ProviderInput) tpmodels.TypeProvider {
+	return providers.Github(input)
+}
 
-// func Discord(config tpmodels.DiscordConfig) tpmodels.TypeProvider {
-// 	return providers.Discord(config)
-// }
+func Google(input tpmodels.ProviderInput) tpmodels.TypeProvider {
+	return providers.Google(input)
+}
 
-// func GoogleWorkspaces(config tpmodels.GoogleWorkspacesConfig) tpmodels.TypeProvider {
-// 	return providers.GoogleWorkspaces(config)
-// }
+func GoogleWorkspaces(input tpmodels.ProviderInput) tpmodels.TypeProvider {
+	return providers.GoogleWorkspaces(input)
+}
 
-// func Google(config tpmodels.GoogleConfig) tpmodels.TypeProvider {
-// 	return providers.Google(config)
-// }
+func CustomProvider(input tpmodels.ProviderInput) tpmodels.TypeProvider {
+	return providers.NewProvider(input)
+}
```

Comment:
Add: 
- Okta
- LinkedIn

Reply:
Done - Okta, Linkedin
BoxyHQ yet to be done